ARM.CMSIS-Compiler.pdsc

```diff
@@ -172,478 +189,242 @@
   </apis>
 
   <components>
-    <!-- File Interface -->
-    <component Cclass="CMSIS-Compiler" Cgroup="File Interface" Csub="Custom" Capiversion="1.0.0" Cversion="1.0.0" custom="1" condition="CortexDevice">
-      <description>Access to #include retarget_fs.h and code template for custom implementation</description>
-      <RTE_Components_h>
-        #define RTE_CMSIS_Compiler_File_Interface        /* Compiler File Interface */
-        #define RTE_CMSIS_Compiler_File_Interface_Custom /* Compiler File Interface: Custom */
-      </RTE_Components_h>
-      <files>
-        <file category="source"  name="template/file_interface/retarget_fs.c" attr="template" select="File Interface Custom Template"/>
-      </files>
-    </component>
-
-    <!-- I/O::File -->
-    <component Cclass="CMSIS-Compiler" Cgroup="I/O" Csub="File" Cvariant="Breakpoint" Cversion="1.0.1" condition="ARMCC CortexDevice IO">
-      <description>Stop program execution at a breakpoint when using file operations</description>
-      <RTE_Components_h>
-        #define RTE_CMSIS_Compiler_IO_File            /* Compiler I/O: File */
-        #define RTE_CMSIS_Compiler_IO_File_BKPT       /* Compiler I/O: File Breakpoint */
-      </RTE_Components_h>
-      <files>
-        <file category="source" name="source/armcc/retarget_io.c"/>
-      </files>
-    </component>
-    <component Cclass="CMSIS-Compiler" Cgroup="I/O" Csub="File" Cvariant="Breakpoint" Cversion="1.0.1" condition="GCC CortexDevice">
-      <description>Stop program execution at a breakpoint when using file operations</description>
-      <RTE_Components_h>
-        #define RTE_CMSIS_Compiler_IO_File            /* Compiler I/O: File */
-        #define RTE_CMSIS_Compiler_IO_File_BKPT       /* Compiler I/O: File Breakpoint */
-      </RTE_Components_h>
-      <files>
-        <file category="source" name="source/gcc/retarget_syscalls.c"/>
-      </files>
-    </component>
-    <component Cclass="CMSIS-Compiler" Cgroup="I/O" Csub="File" Cvariant="File Interface" Cversion="1.0.1" condition="ARMCC CortexDevice FileInterface IO">
-      <description>Redirect file operations to File Interface</description>
-      <RTE_Components_h>
-        #define RTE_CMSIS_Compiler_IO_File            /* Compiler I/O: File */
-        #define RTE_CMSIS_Compiler_IO_File_Interface  /* Compiler I/O: File Interface */
-      </RTE_Components_h>
+    <!-- Core -->
+    <component Cclass="CMSIS-Compiler" Cgroup="CORE" Cversion="1.0.0" condition="ARMCC CortexDevice">
```

Comment:
Why have the Cgroup name all uppercase?

Reply:
- Is abbreviation for standard error stream
- Graphical appearance when displayed with a tool
- Keeping the same casing as in old Arm_Compiler pack to simplify migration

Unfortunately streams are usually represented in all lowercase and that does not look good when we represent components in a GUI tool. The options if we go with abbreviation are:
`Cclass="CMSIS-Compiler" Cgroup="STDERR"`
`Cclass="CMSIS-Compiler" Cgroup="stderr"`
`Cclass="CMSIS-Compiler" Cgroup="Stderr"`

Reply:
I vote for lower case, just as Vlado has explained.

Reply:
Vlado explained:

> all lowercase and that does not look good when we represent components in a GUI tool

I was just referring to `CORE` vs `Core`. For `STDERR` it could also be camel case `StdErr`. There is no perfect solution, it's just a matter of consistency. We use `CORE` for `CMSIS:CORE` as well and changing this to `CMSIS:Core` for cosmetical reason would cause lot of friction.

Reply:
![All upper](https://github.com/ARM-software/CMSIS-Compiler/assets/22930879/335d74e4-f117-4678-9534-125240a37f8b)
vs.
![All lower](https://github.com/ARM-software/CMSIS-Compiler/assets/22930879/4ef80b1f-f3c1-4061-b8f4-2d990737579c)

Reply:
OK, I see. I checked what we have today. And we're using "STDERR" for the Csubs that we are using now. Thus, I change my vote to "leave them upper case".

Reply:
I apologize for the confusion - when I was thinking about Cgroup names I got so fixed to standard streams that I overlooked "CORE".

Reply:
Okay, let